src/main/java/javaslang/collection/Tree.java

```diff
@@ -247,7 +247,7 @@ String toString(Tree<T> tree, int depth) {
                         final String children = tree.getChildren()
                                 .map(child -> toString(child, depth + 1))
                                 .join();
-                        return String.format("\n%s%s%s", indent, value, children);
+                        return String.format("%n%s%s%s", indent, value, children);
```

Comment:
Yes, right. Here we can use `"\n" + indent + value + children`. Could you do that, please?

In general I prefer readability, because the String representations are mostly used for Debug purpose or probably in the future Java 9 REPL :-)

When it comes to high performance, we will use StringBuilder. See also http://stackoverflow.com/a/13815477/1110815